components/Skeleton/Skeleton.jsx

```diff
@@ -1,85 +1,45 @@
 import PropTypes from 'prop-types';
-// eslint-disable-next-line import/no-cycle
-import SkeletonTypes from './SkeletonTypes';
-import { spacing, baseFontSize as defaultBaseFontSize } from '../shared/theme';
-import {
-  SkeletonRect,
-  SkeletonCircle,
-  SkeletonText,
-  SkeletonButton,
-  SkeletonTag,
-} from './sub-components';
+import * as Styles from './Skeleton.styles';
 
-const Skeleton = props => {
-  const { type, size, theme, height, width } = props;
+const Skeleton = ({ type, ...props }) => {
+  switch (type) {
+    case 'circle':
+      return <Styles.Circle {...props} />;
 
-  const rectProps = {
-    height,
-    width,
-  };
-  const circleProps = {
-    height,
-    width,
-  };
+    case 'text':
+      return <Styles.Text {...props} />;
 
-  const textProps = {
-    height,
-    width,
-  };
+    case 'button':
+      return <Styles.Button {...props} />;
 
-  const buttonProps = {
-    size,
-    theme,
-    width,
-  };
+    case 'tag':
+      return <Styles.Tag {...props} />;
 
-  const tagProps = {
-    size,
-    theme,
-    width,
-  };
-
-  return (
-    <>
-      {type === 'rect' && <SkeletonRect {...rectProps} />}
-      {type === 'circle' && <SkeletonCircle {...circleProps} />}
-      {type === 'text' && <SkeletonText {...textProps} />}
-      {type === 'button' && <SkeletonButton {...buttonProps} />}
-      {type === 'tag' && <SkeletonTag {...tagProps} />}
-    </>
-  );
+    default:
+      return <Styles.Rect {...props} />;
+  }
 };
 
 Skeleton.defaultProps = {
   type: 'rect',
+  as: 'div',
   width: '100%',
   height: '14px',
   size: 'medium',
-  theme: {
-    baseFontSize: defaultBaseFontSize,
-    spacing,
-  },
 };
 
 Skeleton.propTypes = {
-  /** Sets the component behavior */
   type: PropTypes.oneOf(['rect', 'circle', 'text', 'button', 'tag']),
+  as: PropTypes.oneOf(['div', 'span']),
   width: PropTypes.string,
-  /** Will affect only types that not uses the prop `size` */
   height: PropTypes.string,
-  /** Will affect only `Tag` and `Button` types.  */
   size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large']),
-  /** Used only for themification.  */
-  theme: PropTypes.shape({
-    spacing: PropTypes.object,
-    baseFontSize: PropTypes.number,
-  }),
 };
 
-// Types
-Skeleton.Button = SkeletonTypes.Button;
-Skeleton.Circle = SkeletonTypes.Circle;
-Skeleton.Text = SkeletonTypes.Text;
-Skeleton.Tag = SkeletonTypes.Tag;
+Skeleton.Rect = props => <Skeleton {...props} />;
+Skeleton.Circle = props => <Skeleton type="circle" {...props} />;
+Skeleton.Text = props => <Skeleton type="text" as="span" {...props} />;
+Skeleton.Button = props => <Skeleton type="button" {...props} />;
+Skeleton.Tag = props => <Skeleton type="tag" {...props} />;
```

Comment:
Honestly, I don't think it will cause any performance problems. The main concern here would be in terms of readability. Even if the user passes additional properties, it will not impact anything if those properties are not being used by the components.

In addition, I raised the idea of moving the codebase to TypeScript. With this migration we can create prop types for each of these variations, which would help to correct use by the user.